### PR TITLE
Update to nextcloud/vue 8

### PR DESCRIPTION
**Pull request by hamza221**
Composer state : I couldn't replicate the unwrapped state we had with the counter
| 1 | 2 | 3 |
|--------|--------|--------|
| image | image | image |

### CODE REVIEW

**Comment by ChristophWurst** [comment]
For a reviewable upgrade, I would suggest to clean up our code from removed components while staying on the old nc/vue, then bump the lib and do the rest of the changes.

Something like
1) A PR against main to replace NcMultiselect with NcSelect
2) A PR against main to migrate NcPopoverMenu
3) ...
4) Bump nc/vue in package.json

https://github.com/nextcloud-libraries/nextcloud-vue/blob/master/CHANGELOG.md#boom-breaking-changes

**Comment by GretaD** [comment]
> For a reviewable upgrade, I would suggest to clean up our code from removed components while staying on the old nc/vue, then bump the lib and do the rest of the changes.
> 
> Something like
> 
> 1. A PR against main to replace NcMultiselect with NcSelect

i have started that one: its almost done, need to rebase it and that its i think: https://github.com/nextcloud/mail/pull/8814

**Comment by GretaD** [comment]
Lets merge this one and close mine https://github.com/nextcloud/mail/pull/8814
This way we avoid more work from hamza to resolve conflicts

**Comment by hamza221** [comment]
- [ ] The only know issue is composer ncSelect, working on fixing it

**Comment by GretaD** [comment]
2 small bugs:

Screenshot from 2023-12-21 15-48-16
Screenshot from 2023-12-21 15-47-46

File: src/views/Setup.vue [hunk 1]
```diff
@@ -71,4 +71,8 @@ export default {
 .mail-empty-content {
 	margin: 0 auto;
 }
+.empty-content {
```
Comment (ChristophWurst) [inline src/views/Setup.vue:74]: nit: BEM and getting rid of the mail-empty-content container

**Review by ChristophWurst — COMMENTED** [review]
Looks sane but didn't test yet

Any strong reason for aliasing NcSelect to SelectVue?

**Comment by hamza221** [comment]
> Any strong reason for aliasing NcSelect to SelectVue?

Select tag already exists in Html, and Not NcSelect just because we're already renaming other components, so to keep it consistent

**Comment by ChristophWurst** [comment]
For consistency with other apps and the upstream name I would prefer NcSelect

**Comment by ChristophWurst** [comment]
> The only know issue is composer ncSelects, working on fixing it

still to do?

**Comment by hamza221** [comment]
> still to do?

Done, but I couldn't mimic the old behaviour tho, might still need improvements.

**Comment by ChristophWurst** [comment]
- [ ] BUG: Half a NcSelect

image

**Comment by nimishavijay** [comment]
Looks good in the screenshots, just not sure about this weird space on the left, the avatar should have equal spacing on the left, top and bottom
image

**Comment by GretaD** [comment]
Bug

- [ ] I Cannot close the Account Settings with the close button or clicking outside. I can close it only with esc
- [ ]  Create calendar and create task - the calendar names are not shown:

Screenshot from 2024-01-09 17-05-40

**Comment by hamza221** [comment]
image
cc @nimishavijay

**Review by GretaD — APPROVED** [review]
tested and rebased, changed 2 small things

**Comment by ChristophWurst** [comment]
https://github.com/nextcloud/mail/pull/9144#issuecomment-1883123348 is still there and a blocker

Bildschirmfoto vom 2024-01-18 09-36-24

**Comment by ChristophWurst** [comment]
- [ ] BUG primary color for active mailboxes

| Before | After |
|--------|--------|
| Bildschirmfoto vom 2024-01-18 09-37-31 | Bildschirmfoto vom 2024-01-18 09-37-13 |

**Comment by ChristophWurst** [comment]
- [ ] BUG app settings

```
[Vue warn]: Duplicate section name found: undefined. Settings navigation sections must have unique section names.

found in

---> <NcAppSettingsSection>
       <NcModal>
         <NcDialog>
           <NcAppSettingsDialog>
             <AppSettingsMenu> at src/components/AppSettingsMenu.vue
               <NcAppNavigation>
                 <Navigation> at src/components/Navigation.vue
                   <NcContent>
                     <Home> at src/views/Home.vue
                       <App> at src/App.vue
                         <Root> 5 vue.runtime.esm.js:4609
``` 

Bildschirmfoto vom 2024-01-18 09-40-59

**Comment by ChristophWurst** [comment]
- [ ] BUG: on mobile, the message actions menu increases font weight for *some* actions

image

**Comment by nimishavijay** [comment]
> image


Much better! 

- [ ] There is still unequal spacing on the left, top and bottom of the avatar, they should all be equal
- [ ] close button should also have a couple of pixels more space on the right

**Comment by GretaD** [comment]
> * [ ]  BUG primary color for active mailboxes
> 
> Before	After
> Bildschirmfoto vom 2024-01-18 09-37-31	Bildschirmfoto vom 2024-01-18 09-37-13

this comes from an index.css that overlaps the active state of app-navigation as well as the envelope list. :deep and !important  on the component doesnt make any difference. 
Screenshot from 2024-01-18 11-16-46

**Comment by ChristophWurst** [comment]
If the color is coming from nc/vue for any active element then I assume this must have a purpose. Let's keep it but make the text readable.

**Comment by GretaD** [comment]
the issue with text on mail was because we force the color on our app. I removed that. The new design is white on active state. See talk as well.
Screenshot from 2024-01-19 16-47-55

**Comment by ChristophWurst** [comment]
- [ ]  BUG

```
[Vue warn]: Invalid prop: custom validator check failed for prop "additionalTrapElements".

found in

---> <NcDialog>
       <NcAppSettingsDialog>
         <AccountSettings> at src/components/AccountSettings.vue
           <RouterLink>
             <NcAppNavigationItem>
               <NavigationAccount> at src/components/NavigationAccount.vue
                 <NcAppNavigation>
                   <Navigation> at src/components/Navigation.vue
                     <NcContent>
                       <Home> at src/views/Home.vue
                         <App> at src/App.vue
                           <Root> vue.runtime.esm.js:4609
```

**Comment by GretaD** [comment]
closed accidentally, because it was mentioned as a fix on the nc/vue pr. sorry about that :)

**Comment by GretaD** [comment]
i tested all what i could. The PR is huge, im sure we will miss something anyway. The most common used actions, work fine.
 If someone has the patience, please test, otherwise, lets merge it :)

**Pull request merged**